modules/widgets/src/fullscreen-widget.tsx

```diff
@@ -0,0 +1,141 @@
+/* global document */
+import {_deepEqual as deepEqual} from '@deck.gl/core';
+import type {Deck, Widget, WidgetPlacement} from '@deck.gl/core';
+import {h, render} from 'preact';
+import {IconButton} from './components';
+
+interface FullscreenWidgetProps {
+  id: string;
+  viewId?: string | null;
+  placement?: WidgetPlacement;
+  /**
+   * A [compatible DOM element](https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullScreen#Compatible_elements) which should be made full screen.
+   * By default, the map container element will be made full screen.
+   */
+  /* eslint-enable max-len */
+  container?: HTMLElement;
+  label?: string;
+  style?: Partial<CSSStyleDeclaration>;
+  className?: string;
+}
+
+export class FullscreenWidget implements Widget<FullscreenWidgetProps> {
+  id = 'fullscreen';
+  props: FullscreenWidgetProps;
+  placement: WidgetPlacement = 'top-left';
+  viewId = null;
+
+  deck?: Deck;
+  element?: HTMLDivElement;
+
+  fullscreen: boolean = false;
+
+  constructor(props: FullscreenWidgetProps) {
+    this.id = props.id || 'fullscreen';
+    this.viewId = props.viewId || null;
+    this.placement = props.placement || 'top-left';
+    props.label = props.label || 'Toggle Fullscreen';
+    props.style = props.style || {};
+    this.props = props;
+  }
+
+  onAdd({deck}: {deck: Deck}): HTMLDivElement {
+    const {style, className} = this.props;
+    const el = document.createElement('div');
+    el.classList.add('deckgl-widget', 'deckgl-widget-fullscreen');
+    if (className) el.classList.add(className);
+    Object.entries(style).map(([key, value]) => el.style.setProperty(key, value));
```

Comment:
This is how Deck applies style: https://github.com/visgl/deck.gl/blob/master/modules/core/src/lib/deck.ts#L734

I have not tested it but the two approaches seem equivalent?
https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/setProperty#alternative_usage

Reply:
I started with that, but it doesn't work for css variables.

`Object.assign(el.style, this.props.style);` results in no variables applied to the middle widget.
<img width="649" alt="Screen Shot 2023-09-09 at 3 19 00 PM" src="https://github.com/visgl/deck.gl/assets/2461547/71a92201-b66e-46c2-a927-2415463e8183">

`el.style.setProperty` has inline styles:
<img width="825" alt="Screen Shot 2023-09-09 at 3 19 23 PM" src="https://github.com/visgl/deck.gl/assets/2461547/b882762c-c246-4010-b426-9e09d0b6c780">

I think we need to change how Deck does it so the theme variables will work.